src/data_source/storage/sql.rs

```diff
@@ -1787,6 +1789,636 @@ impl SqlStorage {
+                            JOIN payload AS p ON h.height = p.height
+                            WHERE h.height = (
+                                SELECT t1.block_height
+                                    FROM transaction AS t1
+                                    WHERE t1.block_height = $1
+                                    ORDER BY (t1.block_height, t1.index) DESC
+                                    OFFSET $2
+                                    LIMIT 1
+                            )
+                            ORDER BY h.height DESC"
+                ),
+                vec![Box::new(*height as i64), Box::new(*offset as i64)],
+            ),
+            explorer::data_source::TransactionIdentifier::Hash(hash) => (
+                format!(
+                    "SELECT {BLOCK_COLUMNS}
+                            FROM header AS h
+                            JOIN payload AS p ON h.height = p.height
+                            WHERE h.height = (
+                                SELECT t1.block_height
+                                    FROM transaction AS t1
+                                    WHERE t1.hash = $1
+                                    ORDER BY (t1.block_height, t1.index) DESC
+                                    LIMIT 1
+                            )
+                            ORDER BY h.height DESC"
+                ),
+                vec![Box::new(hash.to_string())],
+            ),
+        };
+
+        let query_row = self.query_one(&query, params).await?;
+        let block = parse_block::<Types>(query_row)?;
+
+        let txns = block.enumerate().map(|(_, txn)| txn).collect::<Vec<_>>();
+
+        let (offset, txn) = match target {
+            explorer::data_source::TransactionIdentifier::Latest => {
+                txns.into_iter().enumerate().last().unwrap()
+            }
+            explorer::data_source::TransactionIdentifier::HeightAndOffset(_, offset) => {
+                txns.into_iter().enumerate().rev().nth(*offset).unwrap()
+            }
+            explorer::data_source::TransactionIdentifier::Hash(hash) => txns
+                .into_iter()
+                .enumerate()
+                .find(|(_, txn)| txn.commit() == *hash)
+                .unwrap(),
+        };
+
+        Ok(explorer::data_source::TransactionDetailResponse::try_from(
+            (&block, offset, txn),
+        )?)
+    }
+
+    async fn get_explorer_summary(
+        &self,
+    ) -> QueryResult<explorer::data_source::ExplorerSummary<Types>> {
+        let histograms= async {
+            let historgram_query_result = self.query(
+                "SELECT
+                    h.height AS height,
+                    h.timestamp AS timestamp,
+                    h.timestamp - lead(timestamp) OVER (ORDER BY h.height DESC) AS time,
+                    p.size AS size,
+                    (SELECT COUNT(*) AS transactions FROM transaction AS t WHERE t.block_height = h.height) as transactions
```

Comment:
Should we just add `num_txns` to the payload table?

Reply:
We *could*, it would prevent us from having to join, but we will likely not avoid the joins in the future when attempting to pull number of transactions submitted by each rollup.  So while it saves us here, it doesn't prevent all future cases, I believe.

What do you think?

Reply:
I think it's a nice and reasonable thing to have in the db in any case. But not a super high priority, it doesn't need to block this PR